jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -515,3 +521,222 @@ def _update_layer_label_default(self, event={}):
             f'CCW {self.rotation_angle_deg():.2f} ' +
             ('(E-left)' if self.east_left else '(E-right)')
         )
+
+
+def link_image_data(app, link_type='pixels', wcs_fallback_scheme=None, wcs_use_affine=True,
+                    error_on_fail=False):
+    """(Re)link loaded data in Imviz with the desired link type.
+
+    .. note::
+
+        Any markers added in Imviz will need to be removed manually before changing linking type.
+        You can add back the markers using
+        :meth:`~jdaviz.core.astrowidgets_api.AstrowidgetsImageViewerMixin.add_markers`
+        for the relevant viewer(s).
+
+    Parameters
+    ----------
+    app : `~jdaviz.app.Application`
+        Application associated with Imviz, e.g., ``imviz.app``.
+
+    link_type : {'pixels', 'wcs'}
+        Choose to link by pixels or WCS.
+
+    wcs_fallback_scheme : {None, 'pixels'}
+        If WCS linking failed, choose to fall back to linking by pixels or not at all.
+        This is only used when ``link_type='wcs'``.
+        Choosing `None` may result in some Imviz functionality not working properly.
+
+    wcs_use_affine : bool
+        Use an affine transform to represent the offset between images if possible
+        (requires that the approximation is accurate to within 1 pixel with the
+        full WCS transformations). If approximation fails, it will automatically
+        fall back to full WCS transformation. This is only used when ``link_type='wcs'``.
+        Affine approximation is much more performant at the cost of accuracy.
+
+    error_on_fail : bool
+        If `True`, any failure in linking will raise an exception.
+        If `False`, warnings will be emitted as snackbar messages.
+        When only warnings are emitted and no links are assigned,
+        some Imviz functionality may not work properly.
+
+    Raises
+    ------
+    ValueError
+        Invalid inputs or reference data.
+
+    """
+    if len(app.data_collection) <= 1 and link_type != 'wcs':  # No need to link, we are done.
+        return
+
+    if link_type not in ('pixels', 'wcs'):
+        raise ValueError(f"link_type must be 'pixels' or 'wcs', got {link_type}")
+    if link_type == 'wcs' and wcs_fallback_scheme not in (None, 'pixels'):
+        raise ValueError("wcs_fallback_scheme must be None or 'pixels', "
+                         f"got {wcs_fallback_scheme}")
+    if link_type == 'wcs':
+        at_least_one_data_have_wcs = len([
+            hasattr(d, 'coords') and isinstance(d.coords, BaseHighLevelWCS)
+            for d in app.data_collection
+        ]) > 0
+        if not at_least_one_data_have_wcs:
+            if wcs_fallback_scheme is None:
+                if error_on_fail:
+                    raise ValueError("link_type can only be 'wcs' when wcs_fallback_scheme "
+                                     "is 'None' if all data have valid WCS.")

```

Comment:
It looks like we check that at least one loaded image has a WCS, but this error message is saying all images must have WCS. The error message should match the actual check.